File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        // brute force
        // time: O(n^2), space: O(1)
        int[] result = new int[2];
        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                if (nums[i] + nums[j] == target) {
                    return new int[]{i,j};
                }
            }
        }
        return new int[0];
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0;
        int r = numbers.size() - 1;
        while (l < r) {
            int total = numbers[l] + numbers[r];
            if (total == target) {
                break;
            } else if (total > target) {
                r--;
            } else {
                l++;
            }
        }
        // vector<int> result({l + 1, r + 1});

        vector<int> result = {l + 1, r + 1};

        // vector<int> result;
        // result.push_back(l+1);
        // result.push_back(r+1);

        return result;
    }
};
